### ISSUE TITLE
Inspect window kicks the bucket randomly on latest pre-release

### ISSUE THREAD

**Issue by minihazel** [tyfon7/UIFixes#95]
![alt text](https://i.imgur.com/vh5TXnh.png)

I am using Fika with a dedicated client (another PC) + 1 solo player (my PC). Latest versions of everything, except I'm on 3.10.0 and not 3.10.1.

I've had this happen a few times in the past few days. I just updated to latest UI Fixes pre-release (from the Discord thread), and I got this when inspecting a random Colt M4 I looted on Factory. I don't know what else to include, but the only error I saw was from the solo player BepInEx console.

[FullLogOutput.log](https://github.com/user-attachments/files/17997866/FullLogOutput.log)

```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
UIFixes.WeaponModdingPatches.CanModify (EFT.InventoryLogic.Mod item, EFT.InventoryLogic.ItemAddress itemAddress, System.String& error) (at <1a7ecb5d9b5849ff90e0a316f83c72e2>:0)
UIFixes.WeaponModdingPatches.CanModify (EFT.InventoryLogic.Mod item, System.String& error) (at <1a7ecb5d9b5849ff90e0a316f83c72e2>:0)
UIFixes.WeaponModdingPatches+InspectLockedPatch.Postfix (EFT.UI.DragAndDrop.ModSlotView __instance, System.Boolean& ___bool_1, UnityEngine.CanvasGroup ____canvasGroup) (at <1a7ecb5d9b5849ff90e0a316f83c72e2>:0)
(wrapper dynamic-method) EFT.UI.DragAndDrop.ModSlotView.DMD<EFT.UI.DragAndDrop.ModSlotView::method_14>(EFT.UI.DragAndDrop.ModSlotView,bool,EFT.InventoryLogic.Slot)
EFT.UI.DragAndDrop.ModSlotView.SetLocked (System.Collections.Generic.KeyValuePair`2[TKey,TValue] lockedState, EFT.InventoryLogic.Slot slot, EFT.InventoryLogic.Item item) (at <be9862b1593e41e48ad89bd48723d349>:0)
EFT.UI.ItemSpecificationPanel.method_17 (EFT.InventoryLogic.CompoundItem compoundItem) (at <be9862b1593e41e48ad89bd48723d349>:0)
EFT.UI.ItemSpecificationPanel.method_1 () (at <be9862b1593e41e48ad89bd48723d349>:0)
(wrapper dynamic-method) EFT.UI.ItemSpecificationPanel.DMD<EFT.UI.ItemSpecificationPanel::Show>(EFT.UI.ItemSpecificationPanel,ItemContextAbstractClass,System.Collections.Generic.List`1<EFT.MongoID>,TraderControllerClass,EFT.InventoryLogic.CompoundItem[],EFT.SkillManager,WeaponPreviewPool,EFT.UI.ItemUiContext,ItemInfoInteractionsAbstractClass<EFT.InventoryLogic.EItemInfoButton>,EFT.UI.SimpleTooltip,EFT.InventoryLogic.InventoryEquipment)
EFT.UI.ItemUiContext.InitSpecificationPanel (EFT.UI.ItemSpecificationPanel panel, ItemContextAbstractClass itemContext, ItemInfoInteractionsAbstractClass[T] contextInteractions) (at <be9862b1593e41e48ad89bd48723d349>:0)
EFT.UI.InfoWindow.Show (ItemContextAbstractClass itemContext, EFT.InventoryLogic.IItemOwner itemController, TraderClass trader, System.Action onSelectedAction, EFT.UI.ItemUiContext itemUiContext, System.Action onClosedAction, ItemInfoInteractionsAbstractClass[T] contextInteractions) (at <be9862b1593e41e48ad89bd48723d349>:0)
EFT.UI.ItemUiContext+Class2716.method_0 (EFT.UI.InfoWindow window, System.Action setPriority, System.Action onClosed) (at <be9862b1593e41e48ad89bd48723d349>:0)
EFT.UI.ItemUiContext.method_13[T] (ItemContextAbstractClass itemContext, T template, System.Action`3[T1,T2,T3] show) (at <be9862b1593e41e48ad89bd48723d349>:0)
(wrapper dynamic-method) EFT.UI.ItemUiContext.DMD<EFT.UI.ItemUiContext::Inspect>(EFT.UI.ItemUiContext,ItemContextAbstractClass,ItemInfoInteractionsAbstractClass<EFT.InventoryLogic.EItemInfoButton>)
GClass3402.method_12 () (at <be9862b1593e41e48ad89bd48723d349>:0)
(wrapper dynamic-method) GClass3402.DMD<GClass3402::ExecuteInteractionInternal>(GClass3402,EFT.InventoryLogic.EItemInfoButton)
[FullLogOutput.log](https://github.com/user-attachments/files/17997865/FullLogOutput.log)

(wrapper dynamic-method) GClass3405.DMD<GClass3405::ExecuteInteractionInternal>(GClass3405,EFT.InventoryLogic.EItemInfoButton)
ItemInfoInteractionsAbstractClass[T].ExecuteInteraction (T interaction) (at <be9862b1593e41e48ad89bd48723d349>:0)
(wrapper dynamic-method) EFT.UI.DragAndDrop.ItemView.DMD<EFT.UI.DragAndDrop.ItemView::OnClick>(EFT.UI.DragAndDrop.ItemView,UnityEngine.EventSystems.PointerEventData/InputButton,UnityEngine.Vector2,bool)
(wrapper dynamic-method) EFT.UI.DragAndDrop.GridItemView.DMD<EFT.UI.DragAndDrop.GridItemView::OnClick>(EFT.UI.DragAndDrop.GridItemView,UnityEngine.EventSystems.PointerEventData/InputButton,UnityEngine.Vector2,bool)
EFT.UI.DragAndDrop.ItemView.UnityEngine.EventSystems.IPointerClickHandler.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) (at <be9862b1593e41e48ad89bd48723d349>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at <d8b3c578ee264d6096e857a8c2808335>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) (at <d8b3c578ee264d6096e857a8c2808335>:0)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
Class410:UnityEngine.ILogHandler.LogException(Exception, Object)
UnityEngine.EventSystems.EventSystem:Update()
```

**Comment by tyfon7** [comment]
Most likely the null variable is `itemAddress`, which really really shouldn't be null ever. I'll add a null check to prevent the code from puking but the inventory was likely in a bad state.

**Comment by tyfon7** [comment]
Fixed in 3.0.1, forgot to close this
